src/util/markdownToText.ts

```diff
@@ -0,0 +1,291 @@
+ *
+ * NOTE: This function should only be used within the scope of `markdownToText`
+ * in order to ensure that `=scope` is always the first child of thought.
+ * */
+const collapse = (text: string): string => {
+  /**
+   * This function takes in plain-text thoughts and collapses sibling-less scoped
+   * thoughts to the respective parent (if any exists).
+   *
+   * ```
+   * - Parent
+   *   - {" "}
+   *     - =scope
+   *     - Child
+   *     - Child2
+   * - Parent with multiple children that can't be collapsed
+   *   - ${" "}
+   *     - =scope
+   *     - Child
+   *     - Child2
+   *   - Another child
+   * ```
+   *
+   * becomes:
+   *
+   * ```
+   * - Parent
+   *   - =scope
+   *   - Child
+   *   - Child2
+   * - Parent with multiple children that can't be collapsed
+   *   - ${" "}
+   *     - =scope
+   *     - Child
+   *     - Child2
+   *   - Another child
+   * ```
+   */
+
+  // Split lines, will be mutated as we iterate over.
+  const lines = text.split('\n')
+
+  // Iterate over the lines while splicing collapsed thoughts.
+  // This is necessary to account for nested collapsing thoughts.
+  for (let i = 0; i < lines.length; i++) {
+    const line = lines[i]
+    const nextLine = lines[i + 1]
+    const previousLine = lines[i - 1]
+
+    /** Find the indentation of a line. */
+    const getIndent = (line: string) => line.search(/\S|$/) / 2
+
+    // Current line's indentation
+    const indent = getIndent(line)
+
+    if (
+      // Empty thought
+      line.trim() === '-' &&
+      // with a nested scope attribute
+      nextLine?.trimEnd() === `${'  '.repeat(indent + 1)}- =scope` &&
+      // as first child of the previous thought
+      previousLine &&
+      getIndent(previousLine) === indent - 1
+    ) {
+      // Accumulate all lines to outdent
+      const outdentedChildren = [nextLine.slice(2)]
+
+      // Iterate over the rest of the lines and outdent children until we either:
+      // - Find a sibling -> can't collapse; break
+      // - Find a next parent -> no sibling, commit all outdented children; break
+      // - Reach the end of the list -> commit all outdented children
+      for (let j = i + 2; j < lines.length; j++) {
+        const lineIndent = getIndent(lines[j])
+
+        if (lineIndent === indent) {
+          // We found a sibling and can't collapse
+          break
+        } else if (lineIndent < indent || j === lines.length - 1) {
+          // We've found the next parent or reached the end, commit all outdented lines
+          lines.splice(i, 1 + outdentedChildren.length, ...outdentedChildren)
+          break
+        } else {
+          // Outdent the child and add it to the accumulator
+          outdentedChildren.push(lines[j].slice(2))
+        }
+      }
+    }
+  }
+
+  return lines.join('\n')
+}
+
+/** Converts markdown to text compatible with `importText`. */
+export const markdownToText = (markdown: string): string => {
+  const tokens = marked.lexer(markdown.trim())
```

Comment:
I'm getting an error on this line when I try to paste markdown in the UI:

<img width="478" alt="Brave Browser 2024-07-26 07 42 26" src="https://github.com/user-attachments/assets/3ac4bb52-89d3-48ea-8038-7ccfec378023">

Reply:
This is interesting, I can't reproduce. Might there be an issue with node modules and vite cache?

Reply:
Yeah, I thought it looked strange. I'll investigate.

Reply:
It seems that Vite was using an old `marked` version that is a dependency of `typedoc`. I removed `typedoc` in `main` since it never proved useful.

If you could rebase on `main` and regenerate the yarn.lock when you get a chance that would be great.

Reply:
Done! 👍